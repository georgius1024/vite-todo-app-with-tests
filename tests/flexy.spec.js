import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import component from "src/components/flexy.vue";

describe("Flexy component", () => {

  it("mounts without error when true", () => {
    const wrapper = mount(component, {
      props: { condition: true },
    });
    expect(wrapper).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("mounts without error when false", () => {
    const wrapper = mount(component, {
      props: { condition: false },
    });
    expect(wrapper).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders proper DOM', async() => {
    const wrapper = mount(component, {
      props: { condition: true },
    });
    expect(wrapper.html()).not.toContain('NOT true');
    await wrapper.setProps({ condition: false })
    expect(wrapper.html()).toContain('NOT true');
  })
});
