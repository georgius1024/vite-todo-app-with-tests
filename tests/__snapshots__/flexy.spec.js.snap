// Vitest Snapshot v1

exports[`Flexy component > mounts without error when false 1`] = `"<div> Condition is NOT true </div>"`;

exports[`Flexy component > mounts without error when true 1`] = `"<div> Condition is true </div>"`;
